Handle chance of double free scan freq in wpa_supplicant.
When wpas_rm_handle_beacon_req is called, intially
params->freqs is freed and then as part of wpas_clear_beacon_rep_data
same params is getting freed again (data->scan_params.freqs).
---
 wpa_supplicant/rrm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/wpa_supplicant/rrm.c b/wpa_supplicant/rrm.c
index f4fbfa7..b7682c6 100644
--- a/wpa_supplicant/rrm.c
+++ b/wpa_supplicant/rrm.c
@@ -1455,6 +1455,7 @@ void wpas_clear_beacon_rep_data(struct wpa_supplicant *wpa_s)
 
 	eloop_cancel_timeout(wpas_rrm_scan_timeout, wpa_s, NULL);
 	bitfield_free(data->eids);
-	os_free(data->scan_params.freqs);
+	if (data->scan_params.freqs)
+		os_free(data->scan_params.freqs);
 	os_memset(data, 0, sizeof(*data));
 }
-- 
1.9.1

